homeassistant/components/sql/sensor.py

```diff
@@ -328,22 +339,34 @@ def __init__(
         self._attr_unique_id = unique_id
         self._use_database_executor = use_database_executor
         self._lambda_stmt = _generate_lambda_stmt(query)
+        self._attr_has_entity_name = not yaml
         if not yaml and unique_id:
             self._attr_device_info = DeviceInfo(
                 entry_type=DeviceEntryType.SERVICE,
                 identifiers={(DOMAIN, unique_id)},
                 manufacturer="SQL",
-                name=name,
+                name=trigger_entity_config[CONF_NAME].template,
             )
 
+    async def async_added_to_hass(self) -> None:
+        """Call when entity about to be added to hass."""
+        await super().async_added_to_hass()
+        await self.async_update()
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return extra attributes."""
+        return dict(self._attr_extra_state_attributes)
```

Comment:
Why do we need to override `extra_state_attributes`?

Reply:
Because the base class for `ManualTriggerEntity` is otherwise overwriting it trying to render templates.
Here we just need to show what is stored in `_attr_extra_state_attributes` from the refresh in `update()`

Reply:
Ref: https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/template_entity.py#L557

Reply:
Did same in `command_line`
Another thing we could do is to change it in the base class to:
```python
@property
    def extra_state_attributes(self) -> dict[str, Any] | None:
        """Return extra attributes."""
        if hasattr(self, "_attr_extra_state_attributes"):
            return self._attr_extra_state_attributes
        return self._rendered.get(CONF_ATTRIBUTES)
```

Reply:
OK, I see.